lib/conjur/command/users.rb

```diff
@@ -136,6 +136,27 @@ class Conjur::Command::Users < Conjur::Command
       end
     end
 
+    user.desc 'Rotate a user or host API key'
+    user.command :rotate_api_key do |c|
+      c.desc "Login of user who's API key we want to rotate (default: logged in user)"
+      c.flag [:user, :u]
+      c.action do |_global, options, _args|
+        if options.include?(:user)
+          # Make sure we're not trying to rotate our own key with the user flag.
+          if api.username == options[:user]
+            exit_now! 'To rotate your own API key, use this command without the --user flag'
+          end
+          puts api.user(options[:user]).rotate_api_key
+        else
+          username, password = Conjur::Authn.read_credentials
+          new_api_key = Conjur::API.rotate_api_key username, password
+          # Show the new one before saving credentials so we don't lose it on failure.
+          puts new_api_key
```

Comment:
I think there should be an option to turn this off (or on).
Is there a way we could get reasonably certain save_credentials won't fail?